test/sqllogictest/pg_catalog_timezone_names.slt

```diff
@@ -0,0 +1,611 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+mode cockroach
+
+query TTTT colnames,rowsort
+SELECT * FROM pg_catalog.pg_timezone_names
```

Comment:
Won't the output here change twice a year?

Reply:
D'oh! I can't believe I didn't think of that. Yes. In fact much worse than that, because not all timezones observe DST at the same time. 

We don't have a way to mock the nowfn from SLT, do we?

Reply:
We do not. We could add a hook for it here. But it's important to note the hook only works for envd. clusterd doesn't get the nowfn plumbed via the process orchestrator, so even a hook would be quite dubious.

Reply:
I think using NowFn is the wrong approach. What about a test-only system var that, if set, is used for the wall_time on transaction start? That's much more obvious about what it's for and not for, and would work here.

Reply:
That makes sense! I'm not going to have time to rig that up though. If you have bandwidth, can you take that over? If not, we could simply project out the problematic columns from this test for now.

Reply:
🚗 OOC, would/should it be possible to `SELECT ... AS OF` here to get a stable result?

Reply:
Yes, I'll take this over and add that and fix whatever else pops up.

> would/should it be possible to SELECT ... AS OF here to get a stable result

At the moment no because AS OF time and `now()` time are unrelated. `now()` is always current wall clock and AS OF is logical execution time. There is an appealing property though, that if `AS OF` is set, it also sets `now()`. But we're not always in the epoch millis timeline, which is where that gets strange.